python/hsfs/training_dataset.py

```diff
@@ -62,6 +62,7 @@ def __init__(
         from_query=None,
         querydto=None,
         label=None,
+        transformation_function_dto=None,
```

Comment:
I think I mentioned also in the online feature store PR. Don't name the attributes `DTO` they just look ugly on the client side. 

Also I'm a bit confused what's going on. You have both the `_transformation_function_dto` and `_transformation_function`. I doubt you need both. 

We should allow users to configure the transformation function when calling the `create_training_dataset`.

We should also allow users to specify multiple transformation functions, i.e. passing an array instead of a single value.